reconcile/terraform_vpc_peerings.py

```diff
@@ -182,7 +182,7 @@ def build_desired_state_all_clusters(clusters, ocm_map, settings):
                 cluster_info, ocm_map, settings
             )
             desired_state.extend(items)
-        except Exception:
+        except (KeyError, BadTerraformPeeringState, awsapi.MissingARNError):
```

Comment:
I think the only exception to catch here is BadTerraformPeeringState, but I may be wrong.
I would still catch Exception after this block and set error to True. Maybe check here if this is a dry-run, and only set error to True if it is. This way we it is guaranteed that we only exit with success in the very particular case of bad terraform peering state AND dry-run.

Reply:
Not quite. This is a bit complicated (you know best that this is a complex integration!) and I should have explained it properly in the PR description, my apologies. Let me break it one exception class at a time:

### `KeyError`

New clusters have no `id` field, so `ocm.get_aws_infrastructure_access_terraform_assume_role` raises this exception. The first line of this method is:

```python
        cluster_id = self.cluster_ids[cluster]
```

### `BadTerraformPeeringState`

Bad inputs on app-interface lead to either there not being:
* A matching peering for the given provider (`find_matching_peering` returns `None`)
* An AWS account for infrastructure access for the given provider (`aws_account_from_infrastructure_access` returns `None`)
* The requester or accepter VPC IDs not existing or not being accessible in AWS (`get_cluster_vpc_details` returning an empty requester VPC ID)

In these cases, I believe (but may be wrong, this is complicated) that the integration should not fail a PR check.

### `awsapi.MissingARNError`

This is what you've cared about all along. In this case the ARN for the `assume_role` doesn't exist in AWS.